server/services/iam_policy_service.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services
+
+import (
+	"context"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	iampb "google.golang.org/genproto/googleapis/iam/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var missingResource error = status.Error(codes.InvalidArgument, "Missing required argument: resource")
+var missingPolicy error = status.Error(codes.InvalidArgument, "Missing required argument: policy")
+var resourceDNE error = status.Error(codes.NotFound, "Requested resource has no Policy")
+
+// NewIAMPolicyServer returns a new LocationsServer for the Showcase API.
+func NewIAMPolicyServer() iampb.IAMPolicyServer {
+	return &iamPolicyServerImpl{
+		policies: map[string]*iampb.Policy{},
+	}
+}
+
+type iamPolicyServerImpl struct {
+	mu       sync.Mutex
+	policies map[string]*iampb.Policy
```

Comment:
nit: comment that the key is the resource

Reply:
Done.